plugin/src/main/java/com/redhat/jenkins/nodesharingfrontend/SharedNode.java

```diff
@@ -136,6 +138,17 @@ protected void _terminate(TaskListener listener) throws InterruptedException {
                     Thread.currentThread().interrupt();
                 }
             }
+            // If AsyncResourceDisposer is available, process all Disposables first
+            AsyncResourceDisposer disposer =
+                    (AsyncResourceDisposer) Jenkins.getActiveInstance().getAdministrativeMonitor("AsyncResourceDisposer");
```

Comment:
`AsyncResourceDisposer#get()`

Reply:
We can not refer to `resource-disposer` symbols directly if that is an optional dependency.